### DISCUSSION TITLE
Ability to Direct Message (DM)

### DISCUSSION THREAD

**Discussion by rluc99** [jamulussoftware/jamulus#2042] [Ideas]
It would be really useful to be able to direct message people during a jam. 
For instance our band is currently looking for a bass player. If a bass player happens to join us for a few numbers during a public jam session and he is really good, it would be great to be able to ask him if he would be interested in joining us without having to do it "in front of" all the other players on that public jam session.
(I believe JamKazam has that capability.)

**Comment by gilgongo** [comment]
I dimly recall some discussion of adding IRC-like control codes into the chat which would perhaps faciliate this, but I'm not sure where it got to. There are periodically requests to enhance the chat facility in Jamulus, but these usually end up being deprioritised in favour of just using other communication channels (eg email, Whatsapp, etc.) since UDP isn't a very good protocol for it. That's not to say somebody might write a system that integrates with Jamulus in the future though.

**Reply by rluc99** [reply]
@gilongo thanks
I agree that once the contact details between the two parties have been obtained, migrating future conversations to email, Whatsapp, or whatever, would be the aim, it's just about establishing that initial contact (and obtaining the contact details) in a private way - something which I feel people would appreciate/find useful. 
(I wonder how JamKazam achieves it?)

**Comment by ann0see** [comment]
I think in theory you could modify the Chat message sending function in server.cpp. See my fork branch eduTools on how I message people directly

**Reply by rluc99** [reply]
@ann0see 
Please excuse my ignorance, but how do I find "fork branch eduTools"?

**Reply by ann0see** [reply]
If you look under https://github.com/ann0see/jamulus you should find it.

Line https://github.com/ann0see/jamulus/blob/dad992741ec487bb9ca8fbff5764ace5092c4ce1/src/server.cpp#L1484 and similar might be worth a look

**Comment by rluc99** [comment]
I have had some more thoughts about this DM idea. To be useful, the message needs to survive longer than a session (therefore it cannot be tied to the session chat). 
It seems like a separate message system would be needed.
In JamKazam users enter their email address when they first register. During a session you can right-click on a session participant name and a message can be sent directly to them (via email if they go off line). To implement this in Jamulus seems like it would take a lot of work/coding and probably not worth it unless their was large enthusiasm for it!
